test/StressTestMetrics/Program.cs

```diff
@@ -0,0 +1,100 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+
+public class Program
+{
+    private static readonly Meter MyMeter = new Meter("TestMeter", "0.0.1");
+    private static readonly Counter<long> Counter = MyMeter.CreateCounter<long>("counter");
+    private static string[] dimensionValues = new string[] { "DimVal1", "DimVal2", "DimVal3", "DimVal4", "DimVal5", "DimVal6", "DimVal7", "DimVal8", "DimVal9", "DimVal10" };
+    private static Random random = new Random();
+
+    public static void Main(string[] args)
+    {
+        long numberOfMetricWriters = Environment.ProcessorCount;
+        long maxWritesPerWriter = 10000000;
+        var writes = new long[numberOfMetricWriters];
+        using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddSource("TestMeter")
```

Comment:
nit: probably indent on 4? (I'm not sure what's the C# guideline)

```suggestion
            .AddSource("TestMeter")
```